drivers/riello_usb.c

```diff
@@ -1079,14 +1082,33 @@ void upsdrv_updateinfo(void)
 	dstate_setinfo("input.bypass.frequency", "%.2f", DevData.Fbypass/10.0);
 	dstate_setinfo("output.frequency", "%.2f", DevData.Fout/10.0);
 	dstate_setinfo("battery.voltage", "%.1f", DevData.Ubat/10.0);
-	if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+        if (testvar("localcalculation")) {
+                battcharge = ((DevData.Ubat <= 129) && (DevData.Ubat >=107)) ? (((DevData.Ubat-107)*100)/22) : ((DevData.Ubat < 107) ? 0 : 100);
+                battruntime = (DevData.NomBatCap * DevData.NomUbat * 3600.0/DevData.NomPowerKW) * (battcharge/100.0);
+                upsloadfactor = (DevData.Pout1 > 0) ? (DevData.Pout1/100.0) : 1;
+
+                dstate_setinfo("battery.charge", "%u", battcharge );
+                dstate_setinfo("battery.runtime", "%.0f", battruntime/upsloadfactor);
+        }
+        else if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+                upsdebugx(0, "\n If you don't see values for battery.charge and battery.runtime or values are incorrect,"
+                                "try setting \"localcalculation\" flag in \"ups.conf\" "
+                                "options section for this driver!\n");
 		dstate_setinfo("battery.charge", "%u", DevData.BatCap);
-		dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
-	}
-
-	if (DevData.Tsystem < 0xFF)
-		dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
-
+                dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
+        }
+        else {
+                upsdebugx(0, "\n If you don't see values for battery.charge and battery.runtime or values are incorrect,"
```

Comment:
This message seems duplicated... maybe this structure would be better:
````
if (localcalculation) {
...
} else {
    upsdebugx(0, ...)
    if (... < 0xFFFF) {
        ...
    }
}
````

Reply:
Your'e right will change that.

Reply:
updated this, and indentations